src/ITensorAutoHOOT/batch_contraction.jl

```diff
@@ -42,18 +52,55 @@ function Executor(networks::Array, cache::NetworkCache)
   return Executor(cache.net_sums, node_dict)
 end
 
-@non_differentiable Executor(networks::Array, cache::NetworkCache)
+@non_differentiable Executor(networks::Array{Array{ITensor,1},1}, cache::NetworkCache)
+
+function Executor(trees::Array{ContractNode,1}, cache::NetworkCache)
+  return Executor(get_leaves(trees), cache)
+end
+
+@non_differentiable Executor(trees::Array{ContractNode,1}, cache::NetworkCache)
 
 NetworkCache() = NetworkCache([NetworkSum([])], Dict())
 
-function NetworkCache(networks::Array)
+function NetworkCache(networks::Array{Array{ITensor,1},1})
   nodes, node_dict = generate_einsum_expr(networks)
   net_sums = [NetworkSum([go.generate_optimal_tree(n)]) for n in nodes]
   node_index_dict = generate_node_index_dict(node_dict, networks)
   return NetworkCache(net_sums, node_index_dict)
 end
 
-@non_differentiable NetworkCache(networks::Array)
+@non_differentiable NetworkCache(networks::Array{Array{ITensor,1},1})
+
+function NetworkCache(trees::Array{ContractNode,1})
+  nodes, node_dict = generate_einsum_expr(trees)
+  net_sums = [NetworkSum([go.generate_optimal_tree(n)]) for n in nodes]
+  node_index_dict = generate_node_index_dict(node_dict, get_leaves(trees))
+  return NetworkCache(net_sums, node_index_dict)
+end
+
+@non_differentiable NetworkCache(trees::Array{ContractNode,1})
+
+"""
+Generate a cached network, under the constraint that tensors in contract_order will be
+contracted based on the order from Array start to the end.
+"""
+# NOTE: this function is experimental and it hasn't been used in experiments yet.
+function NetworkCache(networks::Array{Array{ITensor,1},1}, contract_order::Array{ITensor})
```

Comment:
This function is not used in any experiments. It's just used for me to test the efficiency of the constrained contraction path algorithm. Unfortunately, it didn't perform very well.